fe/fe-core/src/main/java/com/starrocks/credential/hdfs/HDFSCloudConfiguration.java

```diff
@@ -14,38 +14,95 @@
 
 public class HDFSCloudConfiguration implements CloudConfiguration {
+    private static final Logger LOG = LogManager.getLogger(HDFSCloudConfiguration.class);
+
     private final HDFSCloudCredential hdfsCloudCredential;
+    private String configResources;
+    private String runtimeJars;
 
     public HDFSCloudConfiguration(HDFSCloudCredential hdfsCloudCredential) {
         Preconditions.checkNotNull(hdfsCloudCredential);
         this.hdfsCloudCredential = hdfsCloudCredential;
     }
 
+    public void setConfigResources(String configResources) {
+        this.configResources = configResources;
+    }
+
+    public void setRuntimeJars(String runtimeJars) {
+        this.runtimeJars = runtimeJars;
+    }
+
     public HDFSCloudCredential getHdfsCloudCredential() {
         return hdfsCloudCredential;
     }
 
+    public void addConfigResourcesToConfiguration(String configResources, Configuration conf) {
+        if (Strings.isNullOrEmpty(configResources)) {
+            return;
+        }
+        String[] parts = configResources.split(",");
+        for (String p : parts) {
+            Path path = new Path(StarRocksFE.STARROCKS_HOME_DIR + "/conf/", p);
+            LOG.debug(String.format("Add path '%s' to configuration", path.toString()));
+            conf.addResource(path);
+        }
+    }
+
     @Override
     public void toThrift(TCloudConfiguration tCloudConfiguration) {
-        // TODO
+        tCloudConfiguration.setCloud_type(TCloudType.HDFS);
+        Map<String, String> properties = new HashMap<>();
+        hdfsCloudCredential.toThrift(properties);
+        properties.put(HDFS_CONFIG_RESOURCES, configResources);
+        properties.put(HDFS_RUNTIME_JARS, runtimeJars);
+        properties.put(HDFS_FS_CREDENTIAL_KEY, getCredentialString());
+        if (!properties.isEmpty()) {
+            tCloudConfiguration.setCloud_properties_v2(properties);
+        }
     }
 
     @Override
     public void applyToConfiguration(Configuration configuration) {
-        // TODO
+        hdfsCloudCredential.applyToConfiguration(configuration);
+        configuration.set(HDFS_CONFIG_RESOURCES, configResources);
+        configuration.set(HDFS_RUNTIME_JARS, runtimeJars);
+        addConfigResourcesToConfiguration(configResources, configuration);
+        configuration.setBoolean(HDFS_CONFIG_RESOURCES_LOADED, true);
+        configuration.set(HDFS_FS_CREDENTIAL_KEY, getCredentialString());
     }
 
+    // NOTE(yanz): hdfs credential is quite special. In most cases, people write auth/username/password etc.
+    // in XML file instead of in string literals. So here we use `configResources` and `runtimeJars` as cred string.
+    // And `hdfsCloudCredential` this field is only for broker load, has nothing to do with catalog connector.
     @Override
     public String getCredentialString() {
-        return hdfsCloudCredential.getCredentialString();
+        return "HDFSCloudConfiguration{" +
```

Comment:
I think you need to add HDFSCloudCredential either.

Reply:
OK, Will fix that.